posts/napari-cosmx-basics/using-napari-cosmx.qmd

```diff
@@ -0,0 +1,501 @@
+
+
+```
+
+We can also combine targets with a list of tuples:
+
+```{r}
+#| eval: false
+genes = [('Npy', "orange"), ("Calb1", "cyan")]
+[gem.plot_transcripts(gene = x[0], color = x[1], point_size=10) for x in genes];
+
+```
+
+## Changing transcript transparency
+
+Sometimes transcripts can be stacked on top of each other to the point you cannot
+discern any features. For example (to do). Adjusting the transparents of the layer
+in the GUI only changes the transparency of a single point so that's not feasible.
+But this is possible in interepeter. 
+
+```{r}
+#| eval: false
+gem.viewer.layers['Npy'].opacity = 0.2
+
+```
+
+## Plot all transcripts
+
+This is not advised for resource-limited systems as it plots all transcripts. 
+The method `add_points` plots all the points for a given FOV. If no FOV is
+specified, it will plot all transcripts (this can be taxing on resource-limited
+computers).
+
+```{r}
+#| eval: false
+
+gem.add_points(fov=1)
+```
+
+## Specify individual cell types
+
+You can choose the color of the cell types and which cells to color by supplying
+a dictionary to the 
+
+```{r}
+#| eval: false
+custom_colors = {
+"EPEN":"#AA0DFE",
+"CPEC":"#3283FE",
+"GN":"#85660D",
+"VLMC":"#1C8356"
+}
+
+gem.color_cells('RNA_nbclust_clusters', color=custom_colors)
+
+```
+
+## Color cells with outlines
+
+We can also plot the cell colors as boundaries instead of filled in polygons (@fig-contours).
+
+```{r}
+#| eval: false
+
+# gem.viewer.camera.center = (0.0, -0.5375926704126319, -54.7415680001114)
+# gem.viewer.camera.zoom = 1371.524539264374
+
+gem.color_cells('RNA_nbclust_clusters', contour=2)
+
+```
+
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-contours"
+#| fig-cap: "Cells types (or other metadata items) can be represented as cell boundaries."
+
+knitr::include_graphics("./figures/fig-contours.png")
+```
+
+
+## Changing background color
+
+For some publication styles (_e.g._, posters), turning the background a lighter
+color might be useful. However, when changing the background, some items might be
+more difficult to see (compare @fig-contours with @fig-white).
+
+```{r}
+#| eval: false
+gem.viewer.window.qt_viewer.canvas.background_color_override = 'white'
+```
+
+```{r}
+#| eval: true
+#| echo: false
+#| label: "fig-white"
+#| fig-cap: "Same extent as @fig-contours but with a white background."
+
+knitr::include_graphics("./figures/fig-white.png")
+```
+
+## Scale Bar location
+
+To reposition the scale bar to the bottom left: 
+
+```{r}
+#| eval: false
+
+gem.viewer.scale_bar.position='bottom_left'
+```
+
+# Conclusion
+
+In this post I showed you some of my go-to `napari-cosmx` plugin features that I
+use when analyzing SMI data. In my workflow, I take advantage of the plugin's
+interactivity as well as its underlying functions and methods. This comes in the form
+of "jotting down" settings for reproducibility or fine-tuning an image's aesthetics
```

Comment:
A link to [napari viewer tutorial ](https://napari.org/stable/tutorials/fundamentals/viewer.html)or other useful plugin list might be helpful here as further reading for the audience.

Reply:
Done: 87393bf1d055284e3c736e537160b9fb08f131d1